### ISSUE TITLE
Feature Request: Picture in Picture

### ISSUE THREAD

**Issue by dillonjohnbrown** [brentsimmons/NetNewsWire#1426]
I would love the ability for NetNewsWire to use macOS's native picture-in-picture mode.

**Comment by brentsimmons** [comment]
For… playing videos? Or…?

**Comment by dillonjohnbrown** [comment]
Yeah. I use NetNewsWire to subscribe to YouTube and other video creators. This feature would save me a couple clicks from opening the video in Safari, then opening it in picture-in-picture.

**Comment by Wevah** [comment]
Unfortunately the `WKWebView` properties to allow PiP and fullscreen are private API, which would most likely get us a finger-wag from App Store review. ☹️

**Comment by Wevah** [comment]
That said, it seems pretty trivial to do, otherwise, at least as far as simply making the proper context menu items work.

**Comment by eivindml-solvr** [comment]
This would be really cool, if that also means that YouTube videos can be played in full screen as well. I have moved all my YouTube subscriptions to NNW, and deleted my account. Would be nice with some better ergonomics for playing YouTube videos.

**Comment by marcoskirsch** [comment]
It’s annoying not to be able to make embedded videos (YouTube and others) full screen because “Fullscreen is not supported on your browser”. 
See screenshot: PiP from Safari in the corner but the embedded video from NetNewsWire cannot be full screen.
Should this be a separate Issue?
![IMG_0134](https://user-images.githubusercontent.com/8085768/75610587-32cdb800-5ad8-11ea-865e-d401120d1732.png)

**Comment by vincode-io** [comment]
As @Wevah mentions above, we can’t implement this without using private API’s. We don’t use private because they aren’t supported and can break at any time. Private API also isn’t allowed in apps in the App Store. 

We’ll leave this issue open in case Apple supplies us  with a technology to make this happen. Unfortunately, it just isn’t possible right now.

**Comment by Wevah** [comment]
It looks like at least as of Safari 13.1 WebKit now supports the standard `requestPictureInPicture()` API.

**Comment by SebDanielsson** [comment]
I too use NNW to subscribe to content creators on YouTube and Odysee so this would be an awesome addition😃

**Comment by Krinkle** [comment]
I use NetNewsWire on iOS, with Feedbin as the sync backend. When I follow a YouTube channel in NetNewsWire, there is an inline video player (append by Feedbin as `<iframe>`), and this player supports picture-in-picture on iOS today.

Making the inline player available on iOS/macOS without Feedbin, is tracked at https://github.com/Ranchero-Software/NetNewsWire/issues/3683.

I suppose this issue remains open for making picture-in-picture work for the macOS app, where indeed it seems to not work today.

I'm guessing that when `<iframe src="https://www.youtube.com/embed/T22_rXew72M"></frame>` runs its JavaScript payloads to decode the obscured the HTML5 video that, that through some server-side or JavaScript sniffing, it ends up with a slightly more basic video player on mobile where PiP "just works", and on desktop it activates a bunch more stuff YouTube's own code is effectively breaking PiP by doing so.

**Comment by Wevah** [comment]
> It looks like at least as of Safari 13.1 WebKit now supports the standard `requestPictureInPicture()` API.

Followup: `requestPictureInPicture()` will be denied if the web view isn't configured to allow picture-in-picture.

**Comment by Wevah** [comment]
I've got a branch with this working (as well as adding fullscreen, which *is* public API) if there's any interest in enabling it.
